firmware/ec/test/suites/Test_ocmp_ltc4015.c

```diff
@@ -0,0 +1,833 @@
+    buffer = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_BATTERY_VOLTAGE_LOW, &buffer));
+    TEST_ASSERT_EQUAL(1304, LTC4015_regs[0x01]);
+
+    /* under voltage limit = [VBAT_LO_ALERT_LIMIT] • 128.176(uV) */
+    buffer =  66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_BATTERY_VOLTAGE_HIGH, &buffer));
+    TEST_ASSERT_EQUAL(1304, LTC4015_regs[0x02]);
+
+    /* IBAT_LO_ALERT_LIMIT = (limit*1.46487uV )/RSNSB*/
+    buffer = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_BATTERY_CURRENT_LOW, &buffer));
+    TEST_ASSERT_EQUAL(2054, LTC4015_regs[0x08]);
+
+    /* VIN_LO_ALERT_LIMIT = limit/1.648mV */
+    buffer = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_INPUT_VOLTAGE_LOW, &buffer));
+    TEST_ASSERT_EQUAL(608, LTC4015_regs[0x03]);
+
+    /* IIN_HI_ALERT_LIMIT = (limit*RSNSI)/1.46487uV */
+    buffer = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_INPUT_CURRENT_HIGH, &buffer));
+    TEST_ASSERT_EQUAL(1369, LTC4015_regs[0x07]);
+
+    /* IIN_LIMIT_SETTING = (limit * RSNSI / 500uV) - 1 */
+    inputCurrent = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_INPUT_CURRENT_LIMIT, &inputCurrent));
+    TEST_ASSERT_EQUAL(3, LTC4015_regs[0x15]);
+
+    /* Maximum charge current target = (ICHARGE_TARGET + 1) • 1mV/RSNSB */
+    buffer = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_ICHARGE, &buffer));
+    TEST_ASSERT_EQUAL(2, LTC4015_regs[0x1A]);
+
+    /* vcharge = (VCHARGE_SETTING/105.0 + 2.0)V/cell w/o temp comp.*/
+    buffer = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_VCHARGE, &buffer));
+    TEST_ASSERT_EQUAL(0, LTC4015_regs[0x1B]);
+
+    /* DIE_TEMP_HI_ALERT_LIMIT = (DIE_TEMP – 12010)/45.6°C */
+    buffer = 66539;
+    TEST_ASSERT_EQUAL(true, LTC4015_fxnTable.cb_set_config(
+                        &gbc_pwr_ext_bat_charger,
+                        LTC4015_CONFIG_DIE_TEMPERATURE_HIGH, &buffer));
+    TEST_ASSERT_EQUAL(57746, LTC4015_regs[0x09]);
+}
+
+void test_ocmp_ltc4015_lower_boundary_value(void)
+{
+    int16_t buffer = 0;
+    int16_t inputCurrent = 0;
+
+    /* under voltage limit = [VBAT_LO_ALERT_LIMIT] • 192.264(uV) */
+    buffer = -345;
```

Comment:
Consider using a constant or define for buffer

Reply:
Code is modified as per the changes required.